bin/diagnostics.bash

```diff
@@ -299,36 +299,18 @@ run_diagnostics_namespaced() {
     # -- ConfigMaps --
     printf '%*s\n' "${COLUMNS:-$(tput cols)}" '' | tr ' ' -
     echo "Fetching ConfigMaps <configmaps>"
-    cfg1=$("${here}/ops.bash" kubectl "${cluster}" get pods -n "${namespace}" -o yaml | yq4 '.items[] | select(.status.conditions[] | select(.type != "Ready" and .status != "True")) | [.spec.volumes[].configMap.name]')
-    readarray cfg1_arr < <(echo "$cfg1" | yq4 e -o=j -I=0 '.[]')
-    cfg2=$("${here}/ops.bash" kubectl "${cluster}" get pods -n "${namespace}" -o yaml | yq4 '.items[] | select(.status.conditions[] | select(.type != "Ready" and .status != "True")) | .spec.containers[].envFrom[].configMapRef.name')
-    readarray cfg2_arr < <(echo "$cfg2" | yq4 e -o=j -I=0 '.[]')
-    cfg3=$("${here}/ops.bash" kubectl "${cluster}" get pods -n "${namespace}" -o yaml | yq4 '.items[] | select(.status.conditions[] | select(.type != "Ready" and .status != "True")) | .spec.containers[].env[].ValueFrom.configMapKeyRef.name')
-    readarray cfg3_arr < <(echo "$cfg3" | yq4 e -o=j -I=0 '.[]')
-    cfgs=("${cfg1_arr[@]}" "${cfg2_arr[@]}" "${cfg3_arr[@]}")
-    # shellcheck disable=SC2060
-    # shellcheck disable=SC2207
-    cfgs=($(echo "${cfgs[@]}" | tr [:space:] '\n' | awk '!a[$0]++'))
-
-    for cfg in "${cfgs[@]}"; do
-        if [ "$cfg" == "null" ]; then
-            :
-        else
-            configmap=$(sed -e 's/^"//' -e 's/"$//' <<<"$cfg")
-            "${here}/ops.bash" kubectl "${cluster}" get configmap -n "${namespace}" "$configmap" -o yaml
-        fi
-    done
```

Comment:
Am I reading this correctly that it selects `configmaps` in active use?

Is that distinction not needed? If not then less code is better! 😄

Reply:
It selected configmaps in active use for pods that are not `Ready`, which can have its uses, but I feel like we might be missing important information by not including ConfigMaps making it more difficult for whoever is going to look at the generated diagnostics to debug an issue. But I see merits with both approaches, as this change does generate more diagnostics, which might also be difficult to debug.

Reply:
Filtering locally may be faster than the round trip of asking for more information, so makes sense to err on the side of verbosity.